### PR TITLE
feat: Add mysql support

### PR DESCRIPTION
**Pull request by michaelbukachi**
@DanCardin @oakhan3 Do I need to update the documentation?
Also, I wasn't able to add tests for mysql in `relational/test_generic.py` and `test_rows.py` since mysql doesn't have a default `public` schema. Before we can add tests we'll have create a public schema and make it the default. 
Should I go with this approach?

### CODE REVIEW

**Comment by michaelbukachi** [comment]
It seems the CI is installing new dependencies.

**Comment by DanCardin** [comment]
@michaelbukachi, the existing tests are a bit presumptuous about public being the default schema. I dont think you ought to stray from the default mysql behavior actually.

For `test_rows.py`, if I'm reading it right, there's not really any need for it to have the public schema specified on the model.

For `test_generic.py`, I think you can just test the actually generic bits, i.e. that `tables=` works with the `Model`, `Model.__table__`, and `'tablename'`, which is the more important bit of that file's tests.

The public schema tests was more to verify we could postgres-alike with sqlite, as a lite version that doesn't require docker. So we should probably ultimately alter the sqlite fixture to more flexibly mimic the behaviors as explicit modes (including an unchanged mode).

File: Makefile [hunk 1]
```diff
@@ -9,7 +9,7 @@ install-base:
 	poetry install
 
 install: install-base
-	poetry install -E postgres -E redshift -E mongo -E redis
+	poetry install -E postgres -E redshift -E mongo -E redis -E mysql
 
 ## Test
 test-base:
```
Comment (DanCardin) [inline Makefile:15]: the without-extras tests will probbally require you to update the `setup.cfg` with `mysql` as a new mark, and in this `test-base` add `and not mysql`.

File: src/pytest_mock_resources/container/mysql.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+
+_mysql_container = pytest.fixture("session")(
+    get_container_fn(
+        "_mysql_container",
+        config["image"],
+        {3306: config["port"]},
+        {
+            "MYSQL_DATABASE": config["root_database"],
+            # "MYSQL_USER": config["username"],
```
[the remark this answers is not in the document]
Comment (michaelbukachi) [reply src/pytest_mock_resources/container/mysql.py:66]: mysql images have a super user name `root` if I specify `MYSQL_USER` the user will bound to the specified database thus won't be able create any new databases.

[the remark this answers is not in the document]
Comment (DanCardin) [reply src/pytest_mock_resources/container/mysql.py:66]: aha, then can you remove these commented lines?

**Comment by DanCardin** [comment]
I'm not sure what's going on with poetry/CI. wanna try re-triggering?

**Comment by michaelbukachi** [comment]
> I'm not sure what's going on with poetry/CI. wanna try re-triggering?

I'm also getting the same errors on my own CI account. Btw, do you have a guide for setting up the environment for tests locally? Some tests don't pass when I use make especially `test-parallel`

**Comment by DanCardin** [comment]
I wouldn't expect you to need to do much more than run `poetry install` with the extras (i.e. `make install` inside a virtualenv, have docker running, and run the tests.

Are you seeing the same sort of docker issues you saw on ubuntu in your other issue?

**Comment by DanCardin** [comment]
For what it's worth, I checked out your branch from your fork, and the tests seem to be passing for me.

**Comment by michaelbukachi** [comment]
> For what it's worth, I checked out your branch from your fork, and the tests seem to be passing for me.

Interesting. A question. If the docker container is already running. Will it affect the tests?

**Comment by DanCardin** [comment]
it shouldn't. That's i.e. what the `pmr` utility is doing. Pre-starting the container to avoid the cost of startup time. It just needs to be **some** postgres (or whatever resource) available on the expected port with the expected credentials.

**Comment by michaelbukachi** [comment]
@DanCardin Finally fixed it. The issue was this statement in `test_statements.py`:
```
select table_name from information_schema.views order by table_name
```
Apparently, `mysql` does not limit the views to the current schema. It returns all views in all schemas.
Changing the mysql statement to:
```
        SELECT table_name
        FROM INFORMATION_SCHEMA.views
        WHERE table_name in ('cool_view', 'cool_view_2')
        AND table_schema = (select database())
        ORDER BY table_name
```
solves it. 
An interesting thing to note is that this bug could only be caught by running `pmr mysql` first before running the tests.

**Comment by DanCardin** [comment]
Yea that's a weird sort of edge case that you'd have to be aware of when using (the default) session scope fixture which shares the container. `SELECT datname FROM pg_database` for postgres would similarly show all the things.

I'll create an issue to document this, as it's possible to work around. Unfortunately, it's not really something we can produce good errors for.

**Comment by michaelbukachi** [comment]
@oakhan3 On it.
